whatwg-fetch/whatwg-fetch.d.ts

```diff
@@ -1,98 +1,105 @@
-	credentials?: RequestCredentials;
-	cache?: RequestCache;
-}
+declare type HeadersInit = Headers | string[][] | { [key: string]: string };
+declare class Headers {
+    constructor(init?: HeadersInit);
 
-type RequestContext =
-	"audio" | "beacon" | "cspreport" | "download" | "embed" |
-	"eventsource" | "favicon" | "fetch" | "font" | "form" | "frame" |
-	"hyperlink" | "iframe" | "image" | "imageset" | "import" |
-	"internal" | "location" | "manifest" | "object" | "ping" | "plugin" |
-	"prefetch" | "script" | "serviceworker" | "sharedworker" |
-	"subresource" | "style" | "track" | "video" | "worker" |
-	"xmlhttprequest" | "xslt";
-type RequestMode = "same-origin" | "no-cors" | "cors";
-type RequestRedirect = "follow" | "error" | "manual";
-type RequestCredentials = "omit" | "same-origin" | "include";
-type RequestCache =
-	"default" | "no-store" | "reload" | "no-cache" |
-	"force-cache" | "only-if-cached";
+    append(name: string, value: string): void;
+    delete(name: string): void;
+    get(name: string): string; // | null; (TS 2.0 strict null check)
+    has(name: string): boolean;
+    set(name: string, value: string): void;
 
-declare interface HeadersMap {
-	[index: string]: string;
+    // WebIDL pair iterator: iterable<ByteString, ByteString>
+    entries(): IterableIterator<[string, string]>;
+    forEach(callback: (value: string, index: number, headers: Headers) => void, thisArg?: any): void;
+    keys(): IterableIterator<string>;
+    values(): IterableIterator<string>;
+    [Symbol.iterator](): IterableIterator<[string, string]>;
 }
 
-declare class Headers {
-	constructor(headers?:Headers|HeadersMap)
-	append(name: string, value: string): void;
-	delete(name: string):void;
-	get(name: string): string;
-	getAll(name: string): Array<string>;
-	has(name: string): boolean;
-	set(name: string, value: string): void;
-	forEach(callback: (value: string, name: string) => void): void;
+declare type BodyInit = Blob | ArrayBufferView | ArrayBuffer | FormData /* | URLSearchParams */ | string;
+interface Body {
+    bodyUsed: boolean;
+    arrayBuffer(): Promise<ArrayBuffer>;
+    blob(): Promise<Blob>;
+    formData(): Promise<FormData>;
+    json(): Promise<any>;
+    text(): Promise<string>;
 }
 
-declare class Body {
-	bodyUsed: boolean;
-	arrayBuffer(): Promise<ArrayBuffer>;
-	blob(): Promise<Blob>;
-	formData(): Promise<FormData>;
-	json(): Promise<any>;
-	json<T>(): Promise<T>;
-	text(): Promise<string>;
-}
+declare type RequestInfo = Request | string;
+interface Request extends Body {
+    method: string;
+    url: string;
+    headers: Headers;
 
-declare class Response extends Body {
-	constructor(body?: BodyInit, init?: ResponseInit);
-	static error(): Response;
-	static redirect(url: string, status: number): Response;
-	type: ResponseType;
-	url: string;
-	status: number;
-	ok: boolean;
-	statusText: string;
-	headers: Headers;
-	clone(): Response;
+    type: "" | "audio" | "font" | "image" | "script" | "style" | "track" | "video";
```

Comment:
The other type aliases also align with the WebIDL enums listed in the specification. Let me know if you are willing to make the change, otherwise I can go ahead and merge the PR.